### PR TITLE
Add ip assertions with tests.

### PR DESCRIPTION
**Pull request by Lexty**
Added `Assertion::ip()`, `Assertion::ipv4()` and `Assertion::ipv6()`.
Checking IP address occurs using [filter_var()](http://php.net/manual/function.filter-var.php).
The fourth argument (`$flag`) can be set to: `FILTER_FLAG_IPV4`, `FILTER_FLAG_IPV6`, `FILTER_FLAG_NO_PRIV_RANGE`, `FILTER_FLAG_NO_RES_RANGE` ([doc](http://php.net/manual/filter.filters.flags.php)).

### CODE REVIEW

**Comment by rquadling** [comment]
Can you rebase onto master please.

**Comment by Lexty** [comment]
@rquadling Hope I have done what you asked me. There are no conflicts now.

File: lib/Assert/Assertion.php [hunk 1]
```diff
@@ -1783,6 +1790,61 @@ public static function satisfy($value, $callback, $message = null, $propertyPath
...
+    * (using input_filter/FILTER_VALIDATE_IP).
+    *
+    * @param string      $value
+    * @param string|null $message
+    * @param string|null $propertyPath
+    * @param null|int    $flag
+    *
+    * @link http://php.net/manual/filter.filters.flags.php
+    */
+    public static function ipv4($value, $message = null, $propertyPath = null, $flag = null)
```
Comment (rquadling) [inline lib/Assert/Assertion.php]: Can you make $flag the second parameter as in all other cases, $message and $propertyPath (and where appropriate $encoding) are the last 2 parameters.

**Comment by Lexty** [comment]
Recommit with changes.

**Comment by rquadling** [comment]
Thank you.

**Pull request merged**